nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -215,12 +224,16 @@ public boolean completeUpload(String batchId, String fileIndex, BatchFileInfo fi
         long length = newMetadata.getContentLength();
         String digest = newMetadata.getContentMD5();
         String blobProviderId = transientStoreName; // TODO decouple this
-        Blob blob = new BinaryBlob(new LazyBinary(blobKey, blobProviderId, null), blobKey, filename, mimeType, encoding,
-                digest, length);
+        Binary binary = new LazyBinary(blobKey, blobProviderId, null);
+        Blob blob = new BinaryBlob(binary, blobKey, filename, mimeType, encoding, digest, length);
         Batch batch = getBatch(batchId);
         batch.addFile(fileIndex, blob, filename, mimeType);
 
         return true;
     }
 
+    protected long getMultipartCopyThreshold() {
+        return NON_MULTIPART_COPY_MAX_SIZE + 1;
```

Comment:
Are you sure about the `+ 1`? This value si compared with `>` so a file of size exactly `NON_MULTIPART_COPY_MAX_SIZE + 1` will not use copyFileMultipart while it should as it's beyond the AWS limit.
Or change the `>` in to `>=`.

Reply:
`getMultipartCopyThreshold` should also probably be renamed to make it clear it's the maximum size where we can use non-multipart upload (5GB) — because there's also a a minimum size to use multipart upload (5MB) and I think we're confusing the two.